packages/frontend/core/src/modules/editor-settting/services/editor-setting.ts

```diff
@@ -1,7 +1,25 @@
-import { Service } from '@toeverything/infra';
+import type { Workspace } from '@toeverything/infra';
+import {
+  DocsService,
+  OnEvent,
+  Service,
+  WorkspaceInitialized,
+} from '@toeverything/infra';
 
 import { EditorSetting } from '../entities/editor-setting';
 
+@OnEvent(WorkspaceInitialized, e => e.onWorkspaceInitialized)
 export class EditorSettingService extends Service {
   editorSetting = this.framework.createEntity(EditorSetting);
+
+  onWorkspaceInitialized(workspace: Workspace) {
+    // set default mode for new doc
+
+    workspace.docCollection.slots.docCreated.on(docId => {
```

Comment:
is it possible in this time sequence?
1. doc created
2. doc is set to edgeless mode (using createEdgeless helper)
3. docCreated event emitted and then set to default mode stored in setting

Reply:
I prefer explicitly setting primary mode.

Reply:
docCreated Should always be triggered first